### ISSUE TITLE
How can parts of the content be prevented from being indexed?

### ISSUE THREAD

**Issue by MrGreenTea** [jameslittle230/stork#186]
Hi James, thanks for this great tool!
It works really well, but indexes more content than I want.
I've tried to extend the `html_selector` to only select elements I want to be indexed, but that caused problems with elements being indexed multiple times.

Is there a way to provide a selector blacklist or something similar?

**Comment by jameslittle230** [comment]
Hey @MrGreenTea! Thanks for writing in.

I'm not sure if what you're hoping for is possible today, but let me try putting together an example to see if I understand what you're saying. If I understand you correctly, you want a way to take a page like this:

```html
<main>
  <aside>...</aside>
  <article>
    <p>Article content</p>
    <div class="no-index">This shouldn't be indexed</div>
    <p>More article content</p>
  </article>
</main>
```

and be able to index all the content inside the `article`, but not in the `div.no-index` -- does that sound correct?

If that's the case, I don't think that's possible today.

I'd be interested in building this, but I'd first want to make sure I'm building the right thing. Could you tell me more about what kind of content you have and why you're trying to exclude some of the elements you want indexed?

**Comment by MrGreenTea** [comment]
Yeah that sounds exactly right.

We are currently evaluating this for our documentation.
That includes a lot of examples our tool and its output, which is mostly structured Terminal Output with tables and things like that. Including that in the index and search is more confusing than helpful, so we would like to filter that out.
It also messes up the result list visually.

A smaller thing are the anchor link added to each heading: 
![grafik](https://user-images.githubusercontent.com/9407731/123522847-6581c380-d6c0-11eb-935f-682fedf98efa.png)

That is a small `¶` character that is confusing to see in the results.

Allowing users to exclude content from the index would be very helpful to us in those cases and probably some more that we haven't noticed yet.

**Comment by jameslittle230** [comment]
Makes sense! A selector-based denylist would be useful. I'll do my best to put it in the next release.

Thanks for the feature request! :)

James
